src/app/Console/Commands/Traits/AddonsHelper.php

```diff
@@ -0,0 +1,186 @@
+            }
+        });
+
+        // validate if backpack repo exists
+        if (collect($details)->pluck('url')->contains($backpackRepo)) {
+            $this->note('Backpack repository entry is present in composer.json.');
+
+            return;
+        }
+
+        // Create repositories
+        $this->progressBlock('Creating repositories entry in composer.json');
+
+        $process = new Process(['composer', 'config', 'repositories.backpack', 'composer', $backpackRepo]);
+        $process->run(function ($type, $buffer) use ($backpackRepo) {
+            if ($type === Process::ERR) {
+                // Fallback
+                $composerJson = Str::of(File::get('composer.json'));
+
+                $currentRepositories = json_decode($composerJson)->repositories ?? false;
+
+                $repositories = Str::of(json_encode([
+                    'repositories' => [
+                        'backpack' => [
+                            'type' => 'composer',
+                            'url' => $backpackRepo,
+                        ],
+                    ],
+                ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+                $composerJson = $currentRepositories
+                    // Replace current repositories
+                    ? preg_replace('/"repositories":\s{\r?\n/', $repositories->match("/\{\n\s+([\s\S]+)\n\s{4}\}/m")->append(",\n"), $composerJson)
+                    // Append to the end of the file
+                    : preg_replace("/\r?\n}/", $repositories->replaceFirst('{', ','), $composerJson);
+
+                File::put('composer.json', $composerJson);
+            }
+        });
+
+        $this->closeProgressBlock();
+    }
+
+    /**
+     * Checks for backpack authentication.
+     *
+     * @return void
+     */
+    public function checkForAuthentication()
+    {
+        // Check if auth exists
+        $details = null;
+        $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com']);
+        $process->run(function ($type, $buffer) use (&$details) {
+            if ($type !== Process::ERR && $buffer !== '') {
+                $details = json_decode($buffer);
+            } elseif (File::exists('auth.json')) {
+                $details = json_decode(File::get('auth.json'), true)['http-basic']['backpackforlaravel.com'] ?? false;
+            }
+        });
+
+        // Token exists
+        if ($details) {
+            $this->note('Backpack authentication token is present.');
+
+            return;
+        }
+
+        // Create an auth.json file
+        $this->progressBlock('Creating auth.json file with your authentication token');
+        $this->newLine();
+
+        $this->note('(Find your access token details on https://backpackforlaravel.com/user/tokens)');
+        $username = $this->ask('Access token username');
+        $password = $this->ask('Access token password');
+
```

Comment:
We know we have some "pitfalls" when user adds wrong authentication, and then it's cached and then you need to dabble a little bit in your brain how to fix it (the issues we found in the other PR this one replaced).

I think it would atleast help if we do minimal validation here like, `not empty` and atleast `min:xx` characters (the number configured in backpack application to generate tokens).

It will not solve all the problems but atleast will prevent  "simple" mistakes:
- pressing enter twice;
- not copying correctly from backpack website;
- not pasting correctly on the console;

Reply:
Nice idea, will do it 👌
I'll add simple validations to all the inputs.

Reply:
I agree. But let's make a separate issue please. This doesn't relate to the current PR, it's an unrelated improvement.